.github/workflows/build.yml

```diff
@@ -1,13 +1,25 @@
 name: Build
 
+common-paths-ignore: &common-paths-ignore
+  - 'book/**'
+  - 'doc/**'
+  - _typos.toml
+  - '*.md'
+
 on:
   push:
     branches:
       - main
+    paths-ignore:
+      - <<: *common-paths-ignore
   pull_request:
     branches:
       - main
+    paths-ignore:
+      - <<: *common-paths-ignore
   merge_group:
+    paths-ignore:
+      - <<: *common-paths-ignore
```

Comment:
If it does, it should also not work with PR merging in general I think, as the result of the workflow is "required". Not sure if it will be marked as "skipped" (which in github logic counts as "succeeded") if it was filtered like done here.